contracts/reserve-auction/src/msg.rs

```diff
@@ -56,30 +99,28 @@ pub enum QueryMsg {
     #[returns(AuctionsResponse)]
     AuctionsBySeller {
         seller: String,
-        query_options: Option<QueryOptions<(String, String)>>,
+        query_options: Option<QueryOptions<AuctionKeyOffset>>,
     },
     #[returns(AuctionsResponse)]
     AuctionsByEndTime {
-        end_time: Timestamp,
-        query_options: Option<QueryOptions<(u64, (String, String))>>,
+        end_time: u64,
+        query_options: Option<QueryOptions<AuctionKeyOffset>>,
     },
 }
 
-/// QueryOptions are used to paginate contract queries
 #[cw_serde]
-#[derive(Default)]
-pub struct QueryOptions<T> {
-    /// Whether to sort items in ascending or descending order
-    pub descending: Option<bool>,
-    /// The key to start the query after
-    pub start_after: Option<T>,
-    // The number of items that will be returned
-    pub limit: Option<u32>,
+pub struct ConfigResponse {
+    pub config: Config,
 }
 
 #[cw_serde]
-pub struct ConfigResponse {
-    pub config: Config,
+pub struct HaltManagerResponse {
```

Comment:
When we can return a single type, no need to create a wrapper response type imho.

Reply:
Ah ok agreed